openedx/core/djangoapps/catalog/utils.py

```diff
@@ -82,25 +84,24 @@ def check_catalog_integration_and_get_user(error_message_field):
         return None, catalog_integration
 
 
-def get_programs(site=None, uuid=None, course=None):  # pylint: disable=redefined-outer-name
+def get_programs(site=None, uuid=None, uuids=None, course=None):  # pylint: disable=redefined-outer-name
     """Read programs from the cache.
 
     The cache is populated by a management command, cache_programs.
 
     Keyword Arguments:
         site (Site): django.contrib.sites.models object
         uuid (string): UUID identifying a specific program to read from the cache.
+        uuids (list of string): UUIDs identifying a specific programs to read from the cache.
         course (string): course id identifying a specific course run to read from the cache.
 
     Returns:
         list of dict, representing programs.
         dict, if a specific program is requested.
     """
-    if len([arg for arg in (site, uuid, course) if arg is not None]) != 1:
+    if len([arg for arg in (site, uuid, uuids, course) if arg is not None]) != 1:
```

Comment:
There's a nice pythonic shortcut: `if not any((site, uuid, uuids, course)):`

Reply:
I believe that would verify that a nonzero number is set, missing the "no more than one" case